### PR TITLE
 Fixes #14261 - User API should return correct admin status

### PR DESCRIPTION
**Pull request by ranjan**

### CODE REVIEW

File: app/views/api/v2/users/main.json.rabl [hunk 1]
```diff
@@ -4,6 +4,8 @@ extends "api/v2/users/base"
 
 attributes :firstname, :lastname, :mail, :admin, :auth_source_id, :auth_source_name, :timezone, :locale, :last_login_on, :created_at, :updated_at
 
+node(:admin) { |user| user.admin? }
```
Comment (domcleal) [inline app/views/api/v2/users/main.json.rabl]: Naming this attribute `admin` will override the line above, hiding whether the user has the admin flag or not.  It won't be possible to differentiate between a user the admin flag or a user who's inherited it.

The UI does differentiate in part, by showing the inherited admin status on the list but the edit form would show the value of the user's attribute.

Perhaps `effective_admin` or a separate flag to show only whether admin is inherited also from a group, e.g. `inherits_admin`.

**Comment by domcleal** [comment]
> There were the following issues with the commit message:

There appears to be a leading space in front of the message.

**Comment by ranjan** [comment]
thanks @domcleal 
Are you suggesting  to implement like this

```
attribute :effective_admin
node(:effective_admin) { |u| u.admin? }
```

**Comment by domcleal** [comment]
Yes, but only the second line should be required as there's no `effective_admin` attribute.

**Comment by ranjan** [comment]
@domcleal,  commit updated and test added

**Comment by domcleal** [comment]
[test]

**Comment by dLobatog** [comment]
Merged as 1cf79b15f0a6af381fa1d9f1b6b9306bbb611fd0, thanks @ranjan!

**Pull request closed without merging**